### ISSUE TITLE
docker-compose don't save Usernames and Passwords

### ISSUE THREAD

**Issue by phillipunzen** [filebrowser/filebrowser#1898]
**Description**
I want to create a docker-compose Application.
The container should automatically restart the container with the new image via Watchtower when an update has come out. The container also restarts, it only does not keep the login data and always uses the admin/admin login data after a restart. The user data is not persistent. New users are also deleted.

**Expected behaviour**
I want to have the login data persistent.

**What is happening instead?**
No Error Message.
Filebrowser don't write any data in the filebrowser.db
But the File is mapped to the container

**Additional context**
Server: Debian 11
Docker and Docker-Compose latest Versions

**How to reproduce?**
1. Create the docker-compose file
```
version: '3.1'

networks:
  traefik:
    external: true
  mdb:
    external: true

services:
 filebrowser:
  container_name: filebrowser
  image: filebrowser/filebrowser:latest
  volumes:
    - ./file-data:/srv
    - ./filebrowser.db:/database/filebrowser.db
    - ./settings.json:/config/settings.json
  networks:
    - traefik
    - mdb
  restart: always
  labels:
    - "traefik.enable=true"
    - "traefik.http.routers.manuals.entrypoints=http"
    - "traefik.http.routers.manuals.rule=Host(`manuals.DOMAIN`)"
    - "traefik.http.middlewares.manuals-https-redirect.redirectscheme.scheme=https"
    - "traefik.http.routers.manuals.middlewares=manuals-https-redirect"
    - "traefik.http.routers.manuals-secure.entrypoints=https"
    - "traefik.http.routers.manuals-secure.rule=Host(`manuals.DOMAIN`)"
    - "traefik.http.routers.manuals-secure.tls=true"
    - "traefik.http.routers.manuals-secure.tls.certresolver=http"
    - "traefik.http.routers.manuals-secure.service=manuals"
    - "traefik.http.services.manuals.loadbalancer.server.port=80"
    - "traefik.docker.network=traefik"
    - "traefik.http.routers.manuals-secure.middlewares=secHeaders@file"
```
2. Create the settings.json
```
{
  "port": 80,
  "baseURL": "",
  "address": "",
  "log": "stdout",
  "database": "/database/filebrowser.db",
  "root": "/srv"
}
```

Create the filebrowser.db
`touch filebrowser.db`

Start the docker container
`docker-compose up -d`

**Comment by ramiresviana** [comment]
I recommend you to check your server logs, it provides useful information. Looks like the settings file is not being read, Filebrowser will look for a file called `.filebrowser.{json, toml, yaml, yml}`, check the documentation at https://filebrowser.org/cli/filebrowser

**Comment by badi95** [comment]
I'm having the same issue. After changing the user from admin to my user I cat filebrowser.db and it doesn't have anything in it. Seems like my filebrowser.db file isn't getting updated.

**Comment by badi95** [comment]
```
docker logs filebrowser
2022/06/06 18:36:03 Using config file: /.filebrowser.json
2022/06/06 18:36:03 Listening on [::]:80
2022/06/06 18:36:06 /api/renew: 401 64.132.151.2 <nil>
```
Seem to be using /.filebrowser.json instead of my specified settings.json

**Comment by badi95** [comment]
For a workaround I ran
`docker exec filebrowser cp /config/settings.json /.filebrowser.json`
After restarting the container it starts to save to config/filebrowser.db
This will have to be reissued after each update, but at least it saves the usernames and passwords for now.

**Comment by sandros94** [comment]
As @ramiresviana sugested, a correct setup of the compose environment variables fixes the inconvinience.
```
    environment:
      FB_CONFIG: "/filebrowser.json"
      FB_DATABASE: "/database/filebrowser.db"
```

**Comment by badi95** [comment]
Hmm, I'm still having the issue after setting the environment variables
docker-compose file:
```
   environment:
      - PUID=1000
      - PGID=1000
      - FB_CONFIG="/config/settings.json"
      - FB_DATABASE="/database/filebrowser.db"
```
```
docker exec -it filebrowser env
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
HOSTNAME=5469e2658ea7
TERM=xterm
PUID=1000
PGID=1000
FB_CONFIG="/config/settings.json"
FB_DATABASE="/database/filebrowser.db"
HOME=/root
```
```
docker logs filebrowser
2022/07/05 19:31:05 Using config file: /.filebrowser.json
2022/07/05 19:31:06 Listening on [::]:80
2022/07/05 19:31:12 /api/renew: 401 64.132.151.2 <nil>
2022/07/05 19:31:17 /api/login: 403 64.132.151.2 <nil>
2022/07/05 19:36:29 Caught signal terminated: shutting down.
2022/07/05 19:36:29 accept tcp [::]:80: use of closed network connection
2022/07/05 19:36:30 Using config file: /.filebrowser.json
2022/07/05 19:36:30 Listening on [::]:80
```

**Comment by sandros94** [comment]
> Hmm, I'm still having the issue after setting the environment variables docker-compose file:
> 
> ```
>    environment:
>       - PUID=1000
>       - PGID=1000
>       - FB_CONFIG="/config/settings.json"
>       - FB_DATABASE="/database/filebrowser.db"
> ```
> 
> ```
> docker exec -it filebrowser env
> PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
> HOSTNAME=5469e2658ea7
> TERM=xterm
> PUID=1000
> PGID=1000
> FB_CONFIG="/config/settings.json"
> FB_DATABASE="/database/filebrowser.db"
> HOME=/root
> ```
> 
> ```
> docker logs filebrowser
> 2022/07/05 19:31:05 Using config file: /.filebrowser.json
> 2022/07/05 19:31:06 Listening on [::]:80
> 2022/07/05 19:31:12 /api/renew: 401 64.132.151.2 <nil>
> 2022/07/05 19:31:17 /api/login: 403 64.132.151.2 <nil>
> 2022/07/05 19:36:29 Caught signal terminated: shutting down.
> 2022/07/05 19:36:29 accept tcp [::]:80: use of closed network connection
> 2022/07/05 19:36:30 Using config file: /.filebrowser.json
> 2022/07/05 19:36:30 Listening on [::]:80
> ```

Did you bind the volume for the  settings.json?
In my case, just as a precaution, I renamed the file to filebrowser.json since I wasn't using the filebrowser:s6 image, as per [doc](https://filebrowser.org/installation#docker)

**Comment by GitTworn** [comment]
Anyone found a solution to this yet?

**Comment by sandros94** [comment]
> Anyone found a solution to this yet?

As already documented in this issue, setting a bind to an existing config json is quite enough:

docker-compose.yml:
```
user: 1000:1000
environment:
  FB_CONFIG: "/filebrowser.json"
  FB_DATABASE: "/database/filebrowser.db"
volumes:
  - {PathOrVolumeToFileBrowserRoot}:/srv
  - {PathToFileBrowserRoot}/filebrowser.db:/database/filebrowser.db
  - {PathToFileBrowserRoot}/filebrowser.json:/filebrowser.json
```

filebrowser.json:
```
{
  "port": 80,
  "baseURL": "",
  "address": "",
  "log": "stdout",
  "database": "/database/filebrowser.db",
  "root": "/srv"
}
```

**Comment by GitTworn** [comment]
Thanks!

**Comment by meminens** [comment]
> > Anyone found a solution to this yet?
> 
> As already documented in this issue, setting a bind to an existing config json is quite enough:
> 
> docker-compose.yml:
> 
> ```
> user: 1000:1000
> environment:
>   FB_CONFIG: "/filebrowser.json"
>   FB_DATABASE: "/database/filebrowser.db"
> volumes:
>   - {PathOrVolumeToFileBrowserRoot}:/srv
>   - {PathToFileBrowserRoot}/filebrowser.db:/database/filebrowser.db
>   - {PathToFileBrowserRoot}/filebrowser.json:/filebrowser.json
> ```
> 
> filebrowser.json:
> 
> ```
> {
>   "port": 80,
>   "baseURL": "",
>   "address": "",
>   "log": "stdout",
>   "database": "/database/filebrowser.db",
>   "root": "/srv"
> }
> ```

It's been almost a year but I am struggling to fix this issue. With these settings, every time I recreate the container, the login credentials default to admin/admin. And the filebrowser.db in the bind folder does not get updated at all.

**Comment by j3ko** [comment]
After trying out all the proposed solutions and digging through the code a bit, I have a workaround that works as of 2.26.0:

### Solution

docker-compose:

```docker
  filebrowser:
    image: filebrowser/filebrowser:latest
    container_name: filebrowser
    restart: unless-stopped
    ports:
      - "8181:80"
    volumes:
      - ${MEDIA_ROOT}:/srv
      - ${CONFIG_ROOT}/filebrowser/settings.json:/config/settings.json
      - ${CONFIG_ROOT}/filebrowser/database:/database
    environment:
      - PUID=${PUID}
      - PGID=${PGID}
    command: ["-c", "/config/settings.json"]
```

The important bit here is `command: ["-c", "/config/settings.json"]`

settings.json:

```json
{
  "port": 80,
  "baseURL": "",
  "address": "",
  "log": "stdout",
  "database": "/database/filebrowser.db",
  "root": "/srv"
}
```

### Problem

It looks like a few things are causing issues:

1. When the docker image is built, `.filebrowser.json` is copied into the image and it is being used with no way to override it.  Adding `command: ["-c", "/config/settings.json"]` is cleanest way I can figure out to change this.
1. FB_ environment variables are not respected.  Not sure what the issue is but they just aren't.  Even when you delete `.filebrowser.json`, the FB_ variables are not being used.
1. Binding the volume `- ${CONFIG_ROOT}/filebrowser/database/filebrowser.db:/database/filebrowser.db` will create a directory instead of a file. Not sure why that is.  Maybe it has something to do with how the json is defined?  Either way, removing the `/filebrowser.db` segment works and that's good enough for me.

**Comment by meminens** [comment]
@j3ko Thank you for sharing your settings.json and compose.yml files. However, they don't address my issue. After restarting the container, filebrowser resets the password and other settings. Therefore, I still have the database issue. I see a folder named "database" is created in the docker app data directory. And this folder is empty. Therefore, no database file is present.

**Comment by meminens** [comment]
@j3ko Never mind! I had a typo in my docker compose file. It is working great thank you. I wish the filebrowser maintainers published the corrected docker compose for future users.

**Comment by Conan1231** [comment]
I don't get it, why this Issue is closed.
I spend hours to fix this and even with your "Solution", I can't get this working. Why are there so many tutorials out there explaining this service deployment with docker, but nobody mentions the part with the persistent settings.

Are others not having the same problem?

Edit: Okay last Tutorial I found used the Image: filebrowser/filebrowser:s6 instead of :latest
This was the only thing I changed and I finally got data in my filebrowser.db

**Comment by ofernandofilo** [comment]
**TL;DR**: to fix the problem, I simply forced the use of '/.filebrowser.json' and created the 'filebrowser.db' file manually.

commands executed on a Raspberry Pi 4, 64 bits.

[long list of commands]

    cd /mnt/eusb/ # my default directory, yours will be different.

    mkdir -p /mnt/eusb/fordockers /mnt/eusb/fordockers/filebrowser /mnt/eusb/fordockers/filebrowser/root

    touch /mnt/eusb/fordockers/filebrowser/filebrowser.db

    curl -fsSL "https://github.com/filebrowser/filebrowser/raw/master/docker/root/defaults/settings.json" -o "settings.json"

    cat ./settings.json

    mv ./settings.json /mnt/eusb/fordockers/filebrowser/

    ls -R -lah /mnt/eusb/fordockers

    docker run -d \
    -v /mnt/eusb/fordockers/filebrowser/root:/srv \
    -v /mnt/eusb/fordockers/filebrowser/settings.json:/.filebrowser.json \
    -v /mnt/eusb/fordockers/filebrowser/filebrowser.db:/database/filebrowser.db \
    -e PUID=$(id -u) \
    -e PGID=$(id -g) \
    -p 8080:80 \
    filebrowser/filebrowser:latest

    docker container ls
    # command result:
    #    CONTAINER ID   IMAGE                            COMMAND          CREATED        STATUS                  PORTS                   NAMES
    #    2b893db5e28f   filebrowser/filebrowser:latest   "/filebrowser"   10 hours ago   Up 10 hours (healthy)   0.0.0.0:43210->80/tcp   zen_jackson

    docker logs 2b893db5e28f # pick the ID from previous command
    # command result:
    #    2024/01/07 07:17:01 Using config file: /.filebrowser.json
    #    2024/01/07 07:17:01 Listening on 0.0.0.0:80

done!

[important lines]

    touch /mnt/eusb/fordockers/filebrowser/filebrowser.db                   # create db file manually before running the service
    docker run -d \                                                         # '-d' runs the service in 'detached mode' / background
    -v /mnt/eusb/fordockers/filebrowser/settings.json:/.filebrowser.json \  # forces the use of the default file 

_o/

**Comment by nmincone** [comment]
I've only ever gotten persistent storage to work with this version; https://github.com/hurlenko/filebrowser-docker
